graylog2-server/src/main/java/org/graylog2/alerts/AlertServiceImpl.java

```diff
@@ -189,4 +202,28 @@ public String save(Alert alert) throws ValidationException {
 
         return this.coll.save((AlertImpl)alert).getSavedId();
     }
+
+    @Override
+    public Alert resolveAlert(Alert alert) {
+        if (alert == null || isResolved(alert)) {
+            return alert;
+        }
+
+        final AlertImpl updatedAlert = AlertImpl.create(
```

Comment:
How about adding a `public abstract Builder toBuilder();` to `AlertImpl.java` so you can replace the create call with the following.

```java
final Alert updatedAlert = alert.toBuilder().resolvedAt(Tools.nowUTC()).build();
```